File: server/src/schemas/historico.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class HistoricoItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    criado_em: datetime
    pace_min_km: float | None
    status: str
    nota_geral: float | None


class HistoricoResponse(BaseModel):
    items: list[HistoricoItem]
    total: int
    page: int
    limit: int

File: client/src/api/historico.ts
import { api } from './client'

export interface HistoricoItem {
  id: number
  criado_em: string
  pace_min_km: number | null
  status: string
  nota_geral: number | null
}

export interface HistoricoResponse {
  items: HistoricoItem[]
  total: number
  page: number
  limit: number
}

export interface HistoricoParams {
  page?: number
  limit?: number
}

export async function getHistoricoRequest(
  params: HistoricoParams = {},
): Promise<HistoricoResponse> {
  const { data } = await api.get<HistoricoResponse>('/api/historico-analise', {
    params: {
      page: params.page,
      limit: params.limit,
    },
  })
  return data
}
